src/calculation/CostElement.ts

```diff
@@ -1,23 +1,38 @@
+import { AmountKind } from '~/models/AmountKind';
 import { Cost } from '~/models/Cost';
 import { Currency } from '~/models/Currency';
 import { VatRate } from '~/models/VatRate';
 
 import { roundSum } from './roundSum';
 
 export class CostElement implements Cost {
-  readonly currency: Currency = Currency.PLN;
-
   get grossPrice(): number {
-    return this.netPrice + this.vatSum;
+    return this.amountKind === AmountKind.GROSS
+      ? this.amount
+      : this.amount + this.vatSum;
+  }
+
+  get netPrice(): number {
+    return this.amountKind === AmountKind.NET
+      ? this.amount
+      : this.amount - this.vatSum;
   }
 
   get vatSum(): number {
-    return roundSum((this.vatRate / 100) * this.netPrice);
+    if (this.amountKind === AmountKind.NET) {
+      return roundSum((this.vatRate / 100) * this.amount);
+    } else {
```

Comment:
I would say `else` is not necessary here, you could just return after the first `if`. Leave it do you :smile:

Reply:
Okey, we will use no-else-return rule